Tool to generate yaml representation of the kubectl help text

pwittrock opened the pull request:
**What this PR does / why we need it**:

It generates a yaml representation of the kubectl help so we can generate higher fidelity docs.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35655)

<!-- Reviewable:end -->

pwittrock committed:
Tool to generate yaml representation of the kubectl help text

AdoHe commented:
some verification failed:
```
FAILED   hack/make-rules/../../hack/verify-bazel.sh
hack/make-rules/../../hack/verify-flags-underscore.py
```

AdoHe commented on cmd/genslateyaml/gen_slate_yaml.go:
```diff
@@ -0,0 +1,213 @@
+
+	if len(*outputFile) < 1 {
+		fmt.Printf("Must specify --output-file.\n")
+		os.Exit(2)
```
any special reason to use `2` here? From [http://www.tldp.org/LDP/abs/html/exitcodes.html](http://www.tldp.org/LDP/abs/html/exitcodes.html), seems exit code 2 has special meaning.

AdoHe commented on cmd/genslateyaml/gen_slate_yaml.go:
```diff
@@ -0,0 +1,213 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
```
why add two LICENSE head here?

AdoHe commented:
@pwittrock great tool, I used it to generate the new help doc, looks quite good 👍

pwittrock commented:
:) Thanks Tony.  I plan on addressing your comments today.

pwittrock commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[cmd/genslateyaml/gen_slate_yaml.go, line 44 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/35655#-KV2BBnokIVV4py-FP2L-r1-44:-KVQLcC3V3tNwRjPyIuA:b3g89la) ([raw file](https://github.com/kubernetes/kubernetes/blob/45b602b721084610b7b69a164af1b8db83eaaed6/cmd/genslateyaml/gen_slate_yaml.go#L44)):*
> http://www.tldp.org/LDP/abs/html/exitcodes.html
Nope.  I thought it would be helpful to have different exit codes for different errors and picked 2 arbitrarily.  Fixed.

---

*[cmd/genslateyaml/gen_slate_yaml.go, line 161 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/35655#-KV2BBnokIVV4py-FP2L-r1-161:-KVQLncLbvQZsNHnDKQR:b-886gsx) ([raw file](https://github.com/kubernetes/kubernetes/blob/45b602b721084610b7b69a164af1b8db83eaaed6/cmd/genslateyaml/gen_slate_yaml.go#L161)):*
<details><summary><i>Previously, AdoHe (TonyAdo) wrote…</i></summary>
> why add two LICENSE head here?
</details>
Fixed

---

*[cmd/genslateyaml/gen_slate_yaml.go, line 174 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/35655#-KV2BBnokIVV4py-FP2L-r1-174:-KVQN3jyKYE8KhzXovLQ:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/45b602b721084610b7b69a164af1b8db83eaaed6/cmd/genslateyaml/gen_slate_yaml.go#L174)):*
<details><summary><i>Previously, AdoHe (TonyAdo) wrote…</i></summary>
> can I say a `TopLevelCommand` has a main Command and several SubCommands?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/35655)*
<!-- Sent from Reviewable.io -->

AdoHe commented:
@pwittrock some verification rules failed:
```
FAILED   hack/make-rules/../../hack/verify-bazel.sh
FAILED   hack/make-rules/../../hack/verify-flags-underscore.py
```

pwittrock commented:
Can't find any information about the bazel tests failure.  Updated with the flags fix and will see what happens.

AdoHe commented:
/lgtm @pwittrock

AdoHe commented:
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/35655)*
<!-- Sent from Reviewable.io -->

AdoHe commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.kubernetes.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/35655#-:-KWe_r_AUgbRqjravb9c:bnfp4nl)*
<!-- Sent from Reviewable.io -->

pwittrock commented:
Thanks Tony

AdoHe commented:
/lgtm

pwittrock commented:
@k8s-bot gci gke e2e test this

rmmh commented:
@k8s-bot test this

rmmh commented:
@k8s-bot node e2e test this
